src/sys/unix/uds/listener.rs

```diff
@@ -0,0 +1,251 @@
+use super::socket_addr;
+use crate::event::Source;
+use crate::sys::unix::net::new_socket;
+use crate::sys::unix::UnixStream;
+use crate::unix::SourceFd;
+use crate::{Interests, Registry, Token};
+
+use std::ffi::OsStr;
+use std::os::unix::ffi::OsStrExt;
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+use std::os::unix::net;
+use std::path::Path;
+use std::{ascii, fmt, io, mem};
+
+#[derive(Debug)]
+pub struct UnixListener {
+    inner: net::UnixListener,
+}
+
+/// An address associated with a `mio` specific Unix socket.
+///
+/// This is implemented instead of imported from [`net::SocketAddr`] because
+/// there is no way to create a [`net::SocketAddr`]. One must be returned by
+/// [`accept`], so this is returned instead.
+///
+/// [`net::SocketAddr`]: std::os::unix::net::SocketAddr
+/// [`accept`]: #method.accept
+pub struct SocketAddr {
```

Comment:
Let's **not** do it in this PR, but as a follow up it would be nice to move this type to a separate file.